ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -192,31 +231,50 @@ class LogFilesViewController : UITableViewController, Themeable {
 
 		}
 		let shareableFileName = "ownCloud_" + time + ".log.txt"
+		
+		if makeNamedCopy {
+			let shareableLogURL = FileManager.default.temporaryDirectory.appendingPathComponent(shareableFileName)
 
-		let shareableLogURL = FileManager.default.temporaryDirectory.appendingPathComponent(shareableFileName)
+			do {
+				if FileManager.default.fileExists(atPath: shareableLogURL.path) {
+					try FileManager.default.removeItem(at: shareableLogURL)
+				}
 
-		do {
-			if FileManager.default.fileExists(atPath: shareableLogURL.path) {
-				try FileManager.default.removeItem(at: shareableLogURL)
+				try FileManager.default.copyItem(atPath: logRecord.url.path, toPath: shareableLogURL.path)
+			} catch {
 			}
-
-			try FileManager.default.copyItem(atPath: logRecord.url.path, toPath: shareableLogURL.path)
-		} catch {
+		
+			completionHandler(shareableLogURL, shareableFileName, {
+				do {
+					try FileManager.default.removeItem(at: shareableLogURL)
```

Comment:
@felix-schwarz just use `try?` or do something in `catch` block: add log entry for instance?

Reply:
Indeed, `try?` is a better choice here. Changing that.